src/Entity/OgMembership.php

```diff
@@ -157,6 +159,95 @@ public function getType() {
   }
 
   /**
+   * Set the group's role's for the current user group membership's.
+   *
+   * @param $roles_ids
+   *   List of og roles ids.
+   *
+   * @return OgMembership
+   */
+  public function setRoles($roles_ids) {
+    $this->set('roles', $roles_ids);
+
+    return $this;
+  }
+
+  /**
+   * Adding a role to the user membership.
+   *
+   * @param $role_id
+   *   The OG role ID.
+   *
+   * @return OgMembership
+   */
+  public function addRole($role_id) {
+    $rids = $this->getRoles(TRUE);
+    $rids[] = $role_id;
+
+    return $this->setRoles(array_unique($rids));
+  }
+
+  /**
+   * Revoking a role from the user's membership's.
+   *
+   * @param $role_id
+   *   The OG role ID.
+   *
+   * @return OgMembership
+   */
+  public function revokeRole($role_id) {
+    $rids = $this->getRoles(TRUE);
+    $key = array_search($role_id, $rids);
+    unset($rids[$key]);
+
+    return $this->setRoles(array_unique($rids));
+  }
+
+  /**
+   * Get all the referenced OG roles.
+   *
+   * @param $return_ids
+   *   Determine if we need to return the role IDs or the roles object's.
+   *
+   * @return OgRole[]
+   */
+  public function getRoles($return_ids = FALSE) {
+    $roles = $this->get('roles')->referencedEntities();
+
+    if ($return_ids) {
+      $rids = [];
+
+      foreach ($roles as $role) {
+        $rids[] = $role->id();
+      }
+
+      return $rids;
+    }
+
+    return $roles;
+  }
+
+  /**
+   * Checks if the user has a permission inside the group.
+   *
+   * @param $permission
+   *   The name of the permission.
+   *
+   * @return bool
+   */
+  public function hasPermission($permission) {
+    $roles = $this->getRoles();
+
```

Comment:
return array_filter($this->getRoles(), function ($role) use ($permission) {return $role->hasPermission($permission);} )